internal/vsp/vsp.go

```diff
@@ -242,14 +242,12 @@ func (c *Client) ProcessWithPolicy(ctx context.Context, ticketHash *chainhash.Ha
 
 	vspTicket, err := c.Wallet.VSPTicketInfo(ctx, ticketHash)
 	if err != nil && !errors.Is(err, errors.NotExist) {
-		log.Error(err)
-		return nil
+		return err
 	}
 
-	switch udb.FeeStatus(vspTicket.FeeTxStatus) {
-	// If VSPTicket has been started or errored then attempt to create a new fee
-	// transaction, submit it then confirm.
-	case udb.VSPFeeProcessStarted, udb.VSPFeeProcessErrored:
+	// VSPTicket info doesn't exists yet so we need to do all the basic
+	// processing.
+	if vspTicket == nil {
```

Comment:
There's now a chunk of non-trivial code that is (and will always need to be) duplicated across the `vspTicket == nil` and `case udb.VSPFeeProcessStarted, udb.VSPFeeProcessErrored` cases, which (while correct now) could trigger a bug in the future if one of the instances is modified by the other is not.

I suggest simplifying this to:

```go
feeStatus := udb.VSPFeeProcessStarted // Will be used if the ticket isn't registered to the vsp yet.
if vspTicket != nil {
  feeStatus = vspTicket.FeeTxStatus
}

switch (feeStatus) {
case udb.VSPFeeProcessStarted, udb.VSPFeeProcessErrored:
  // ...
// other cases
}
```